docs/intro-to-lora.md

```diff
@@ -1,139 +1,414 @@
+
+1. After this, the gateway shows as **Connected** in the TTN console.
+
+    <span class="images">![Connected!](assets/ttn7.png)<span>Connected!</span></span>
+
+### 3. Building a device
+
+This section explains how to build a device that can send sensor data over the LoRa network. For example, you can create a motion sensor using a [PIR sensor](https://www.adafruit.com/products/189) (less than 10 euros). Of course, you can use any other sensor.
+
+<span class="images">![nRF51-DK, LoRa shield and a PIR sensor](assets/lora6.jpg)<span>PIR sensor hooked up to a Nordic Semiconductor nRF51-DK with a SX1276 LoRa shield</span></span>
+
+#### 3.1 - Some notes on writing firmware
+
+##### Restrictions on sending data 
+
+You cannot send data constantly because of spectrum regulations. Although the spectrum that LoRa uses is unlicensed, it is regulated. For example, in Europe, there are duty cycle limitations of 1% - meaning you can only send 1% of the time. In the US, there's dwell time, which requires you to wait at least 400 ms between transmissions. If you violate these regulations, your data transmission fails. How fast you are allowed to send data depends on the spread factor you use. With a higher spread factor, it takes longer to send a message - though the chance that a gateway receives it increases. However, you need to wait longer before you can send data again. During development, you can set the spread factor to SF7 (the lowest), so you can send every 6-7 seconds.
+
+LoRaWAN has a feature called Adaptive Data Rating (ADR), through which the network can control the spread factor. You probably want this enabled.
+
+##### Blocked pins
+
+A disadvantage of the SX1272 and SX1276 LoRa shields is that they block a lot of pins. You can solder new headers on the back of the shield to add new peripherals, or use a microcontroller such as the nRF51-DK or a NUCLEO board that has more pins available than just the Arduino headers.
+
+#### 3.2 - Registering the device on The Things Network
+
+LoRaWAN uses an end-to-end encryption scheme that uses two session keys. The network server holds one key, and the application server holds the other. (In this tutorial, TTN fulfils both roles). These session keys are created when the device joins the network. For the initial authentication with the network, the application needs its device EUI, the EUI of the application it wants to join (referred to as the application EUI) and a preshared key (the application key).
+
+The device EUI and application EUI are globally unique identifiers. You can buy a block of EUIs from the [IEEE](http://standards.ieee.org/develop/regauth/tut/eui.pdf). If you're using a module, it may have an EUI printed on the module. If you're using a radio shield, you can use an EUI from The Things Network's block.
```

Comment:
Query: "If you're using a module, it may have an EUI printed on the module." Should "module" be in there twice?